content/introducing-user-roles-authorization-with-supertokens/index.md

```diff
@@ -0,0 +1,200 @@
+
+![AuthN to AuthZ](./authn-to-authz.png)
+
+### What are user roles and why are they important for user authentication?
+
+User Roles is built based on principles of roles-based access control (RBAC).
+
+RBAC is a security model that restricts access to resources based on the roles that users have within an organization. Users are assigned to specific roles and those roles are granted permissions to access certain resources.
+
+Let's try building a blogging app with user roles.  
+
+We'd want to have a `regular users` role that allows `regular users` to read all blog posts, but only edit or delete posts created by themselves. We'd also want to have an `admin` role where `admins` can create, edit, and delete all blog posts.
+
+In formal notation, the actions that can be taken can be represented as strings like:
+
+-   Read all blogs: `read:all`
+
+-   Delete all blogs: `delete:all`
+
+-   Delete blogs created by self: `delete:self`
+
+-   Edit all blogs: `edit:all`
+
+-   Edit blogs created by self: `edit:self`
+
+These are known as permissions. In a grid form, the roles and permissions for our example app will look like this:
+
+![Permission Grid](./permission_grid.png)
+
+
+The "edit:all" permission implies "edit:self" as well. Same goes for the "delete" permission.
+
+To learn more about RBAC, check out our [blog post going in depth on RBAC](http://supertokens.com/blog/roles-based-access-control-with-supertokens).
+
+### User Roles and SuperTokens
+
+We built user roles with simplicity in mind. Our goal was to build secure authorization that can be set up in the span of minutes, not hours. As a result, our default user roles product is perfect for most non-complex use cases. For power users, we've left plenty of room to work with a product such as [permit.io](http://permit.io/) to extend our core permission feature set.
+
+Today, you can create new roles and permissions, assign them to users, and protect API and website routes with SuperTokens. Let's go over each one at a high level.
+
+Create new roles and permissions - With SuperTokens, roles and permissions are simple strings that can be assigned to users. That means you can name your role anything from `user` to `super-admin-i-have-all-the-permissions`.
+
+Assign roles to users - Once you've created the role, you can assign roles to users before or after sign up. In fact, you can even assign multiple roles to a single user, so you could give someone both `regular-user` and `admin` permissions.
+
+Product API and website routes - Now to verify if a user has proper permissions, we need to verify that a session exists and validate that the roles/permissions saved in the access token payload have the appropriate value. Getting the user role is simple with `UserRoleClaim` and since our roles are strings, verification is as simple as a string comparison.
```

Comment:
```suggestion
- Product API and website routes - Now to verify if a user has proper permissions, we need to verify that a session exists and validate that the roles/permissions saved in the access token payload have the appropriate values. Fetching the user's role from their session is simple, and since our roles are strings, verification is as simple as a string comparison.
```